addons/mail/static/src/models/thread/thread.js

```diff
@@ -265,79 +272,110 @@ function factory(dependencies) {
-         *   or chat window, depending on whether discuss is open or not.
-         * @param {string} [param0.autoselectChatWindowMode]
          * @param {string} param0.name
-         * @param {integer} [param0.partnerId]
-         * @param {string} [param0.public]
-         * @param {string} param0.type
-         */
-        static async createChannel({
-            autoselect = false,
-            autoselectChatWindowMode,
-            name,
-            partnerId,
-            public: publicStatus,
-            type,
-        }) {
+         * @param {string} [param0.privacy]
+         * @returns {mail.thread} the created channel
+         */
+        static async createChannel({ name, privacy }) {
             const device = this.env.messaging.device;
-            // TODO FIX: potential duplicate chat task-2276490
             const data = await this.env.services.rpc({
                 model: 'mail.channel',
-                method: type === 'chat' ? 'channel_get' : 'channel_create',
-                args: type === 'chat' ? [[partnerId]] : [name, publicStatus],
+                method: 'channel_create',
+                args: [name, privacy],
                 kwargs: {
                     context: Object.assign({}, this.env.session.user_content, {
+                        // optimize the return value by avoiding useless queries
+                        // in non-mobile devices
                         isMobile: device.isMobile,
                     }),
-                }
+                },
             });
-            const thread = this.insert(this.convertData(data));
-            if (autoselect) {
-                thread.open({ chatWindowMode: autoselectChatWindowMode });
-            }
+            return this.env.models['mail.thread'].insert(
+                this.env.models['mail.thread'].convertData(data)
+            );
         }
 
         /**
-         * Join a channel. This channel may not yet exists in the store.
+         * Creates a chat, which is a special kind of thread on model
+         * 'mail.channel' with multiple members. An existing chat might be
+         * returned if one already exists with the given partners.
+         *
+         * This method is a low level wrapper for the `channel_get` RPC.
+         * `openChat` is preferable in business code because it will avoid the
+         * RPC if the chat already exists.
          *
          * @static
-         * @param {integer} channelId
-         * @param {Object} [param1={}]
-         * @param {boolean} [param1.autoselect=false]
-         */
-        static async joinChannel(channelId, { autoselect = false } = {}) {
-            let channel = this.find(thread =>
-                thread.id === channelId &&
-                thread.model === 'mail.channel'
-            );
-            if (channel && channel.isPinned) {
+         * @param {Object} param0
+         * @param {integer[]} param0.partnerIds
+         * @param {boolean} [param0.pinForCurrentPartner]
```

Comment:
Missing default? It passes `undefined` to RPC...

Reply:
The param is not required on the server but has a default value. The goal is to use the default from the server.
Passing `undefined` to RPC is equivalent to not setting the key, both it does not actually send `undefined`.